pkg/tsdb/sqleng/sql_engine.go

```diff
@@ -314,7 +314,8 @@ func (e *DataSourceHandler) executeQuery(query backend.DataQuery, wg *sync.WaitG
 
 	// If no rows were returned, no point checking anything else.
 	if frame.Rows() == 0 {
-		queryResult.dataResponse.Frames = data.Frames{}
+		frame.Fields = []*data.Field{}
```

Comment:
I'd love to see a comment on this why this is needed but other than that good job!

Reply:
Great point; I agree; a change like this is hard to understand without a little context.

Reply:
Updated with annotation snippet.